packages/ee/ui-platform/src/lib/components/smtp-settings/smtp-settings.component.ts

```diff
@@ -0,0 +1,102 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { FlagService, PlatformService } from '@activepieces/ui/common';
+import { ActivatedRoute } from '@angular/router';
+import { Platform } from '@activepieces/ee-shared';
+import { BehaviorSubject, Observable, catchError, tap } from 'rxjs';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+interface SmtpForm {
+  smtpHost: FormControl<string>;
+  smtpPort: FormControl<number>;
+  smtpUser: FormControl<string>;
+  smtpPassword: FormControl<string>;
+  smtpSenderEmail: FormControl<string>;
+  smtpUseSSL: FormControl<boolean>;
+}
+@Component({
+  selector: 'app-smtp-settings',
+  templateUrl: './smtp-settings.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SmtpSettingsComponent implements OnInit {
+  smtpSettingsForm: FormGroup<SmtpForm>;
+  showPoweredByAp$: Observable<boolean>;
+  loading$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  saving$?: Observable<void>;
+  constructor(
+    private fb: FormBuilder,
+    private platformService: PlatformService,
+    private route: ActivatedRoute,
+    private flagService: FlagService,
+    private matSnackbar: MatSnackBar
+  ) {
+    this.smtpSettingsForm = this.fb.group({
+      smtpHost: this.fb.control('', {
+        nonNullable: true,
+        validators: [Validators.required],
+      }),
+      smtpPort: this.fb.control(1234, {
+        nonNullable: true,
+        validators: [Validators.required],
+      }),
+      smtpUser: this.fb.control('', {
+        nonNullable: true,
+        validators: [Validators.required],
+      }),
+      smtpPassword: this.fb.control('', {
+        nonNullable: true,
+        validators: [Validators.required],
+      }),
+      smtpSenderEmail: this.fb.control('', {
+        nonNullable: true,
+        validators: [Validators.required],
+      }),
+      smtpUseSSL: this.fb.control(false, {
+        nonNullable: true,
+        validators: [Validators.required],
+      }),
+    });
+    this.showPoweredByAp$ = this.flagService.getShowPoweredByAp();
+  }
+  ngOnInit(): void {
+    const platform: Platform = this.route.snapshot.data['platform'];
+    this.smtpSettingsForm.patchValue(platform);
```

Comment:
I think in the backend It doesn't return smtpPassword for security reasons.

When you removed the platform owner check in get endpoint of platform, I didn't add smtp password as it can be accessed by any platform member even if not owner.

I am not sure what's do in that case, but will leave it up to you.